go/interpreter/lfvm/interpreter_intergration_test.go

```diff
@@ -0,0 +1,689 @@
+				mock.EXPECT().AccountExists(gomock.Any()).Return(true)              // non-empty account
+				mock.EXPECT().GetBalance(gomock.Any()).Return(tosca.Value(callGas)) // enough balance to transfer value
+				mock.EXPECT().Call(gomock.Any(), gomock.Any()).Return(tosca.CallResult{}, nil)
+			},
+			gas: cost(gas(
+				700 + // static
+					3 + // mem-expansion
+					9000 - 2300 + // call-stipend
+					700 + // value-transfer
+					2300, // call-stipend return
+			)),
+		},
+
+		{
+			nameOverride: "CALL EIP-2929",
+			code:         []Instruction{{CALL, 0}},
+			stack: stackWithValues([]tosca.Word{
+				retSize,
+				retOffset,
+				argsSize,
+				argsOffset,
+				value,
+				address,
+				callGas,
+			}, 1),
+			revisionConstraint: validFrom(tosca.R09_Berlin),
+			mockCalls: func(mock *tosca.MockRunContext) {
+				mock.EXPECT().AccountExists(gomock.Any()).Return(true)              // non-empty account
+				mock.EXPECT().GetBalance(gomock.Any()).Return(tosca.Value(callGas)) // enough balance to transfer value
+				mock.EXPECT().IsAddressInAccessList(gomock.Any()).Return(true)      // warm
+				mock.EXPECT().Call(gomock.Any(), gomock.Any()).Return(tosca.CallResult{}, nil)
+			},
+			gas: cost(gas(
+				0 + // static
+					3 + // mem-expansion
+					100 + // warm access
+					9000 - 2300 + // call-stipend
+					700 + // value-transfer
+					2300, // call-stipend return
+			)),
+		},
+
+		{
+			code: []Instruction{{DELEGATECALL, 0}},
+			stack: stackWithValues([]tosca.Word{
+				retSize,
+				retOffset,
+				argsSize,
+				argsOffset,
+				address,
+				callGas,
+			}, 1),
+			revisionConstraint: validBefore(tosca.R09_Berlin),
+			mockCalls: func(mock *tosca.MockRunContext) {
+				mock.EXPECT().Call(gomock.Any(), gomock.Any()).Return(tosca.CallResult{}, nil)
+			},
+			gas: cost(gas(
+				700 + // static
+					3 + // mem-expansion
+					700, // value-transfer
+			)),
+		},
+	}...)
+
+	return tests
+}
+
+// FIXME: this integration test does not cover all the opcodes. this test will fail
+func testInterpreter_TestCasesAreComplete(t *testing.T) {
+
+	// This test is here to ensure that all opcodes are covered by the test cases.
+	// If an opcode is not covered, the test will fail and the missing opcode will be printed.
+	testedOps := map[OpCode]map[tosca.Revision]bool{}
+	for _, test := range generateTestCases() {
+		for _, instr := range test.code {
+			if _, ok := testedOps[instr.opcode]; !ok {
+				testedOps[instr.opcode] = map[tosca.Revision]bool{}
+			}
+			forEachSupportedRevision(t, test, func(t *testing.T, revision tosca.Revision) {
+				testedOps[instr.opcode][revision] = true
+			})
+		}
+	}
+
+	re := regexp.MustCompile(`^[A-Z0-9_]+$`)
+	for i := 0; i < numOpCodes; i++ {
+		op := OpCode(i)
+		if !re.MatchString(op.String()) {
+			continue
+		}
+
+		if _, ok := testedOps[op]; !ok {
+			t.Errorf("opcode %v is not tested", op)
+		} else {
+			// FIXME: false negatives, some ops just wont exist in older revisions
```

Comment:
these exceptions are few enough that they could be listed there and this test "fixed" ? I guess this could also be a small follow-up